scripts/deploy.ts

```diff
@@ -1,30 +1,28 @@
 import hre from "hardhat";
 
 async function main() {
-    const [deployer, gameOwner] = await hre.ethers.getSigners();
+    const [deployer] = await hre.ethers.getSigners();
 
     // Governance Token deployment
     console.log("Deploying Governance contract with the account:", deployer.address);
     console.log("Account balance:", (await deployer.getBalance()).toString());
 
     const governanceTokenFactory = await hre.ethers.getContractFactory("GovernanceToken");
 
+    const addressList = [process.env.EXCHANGES_WALLET_ADDRESS, process.env.PLAYANDEARN_WALLET_ADDRESS,
+        process.env.SOCIAL_WALLET_ADDRESS, process.env.TEAM_WALLET_ADDRESS, process.env.TREASURY_WALLET_ADDRESS, process.env.ADVISORS_WALLET_ADDRESS];
+
     const governanceToken = await governanceTokenFactory.deploy(
-        8000000000000000, "MEE Governance Token", 18, "MEE", process.env.GAME_OWNER_ADDRESS, {gasLimit: 6e6});
+        8000000000000000, "MEE Governance Token", 18, "MEE", process.env.GAME_OWNER_ADDRESS,
```

Comment:
Bug: I think we supply 3 billion tokens, not 8. I suggest you put the supply in the constructor, not a factory parameter if possible.

Reply:
The number is also strange, it doesn't seem like 8*10^(18+9), i.e. 8 billion with our 18 decimals, or just 8 billion.

And it should be 3. :)

Reply:
There is also conflicting code in deploy.ts, so which one is the right one?

```
    const governanceToken = await governanceTokenFactory.deploy(
        "MEE Governance Token", 18, "MEE", process.env.GAME_OWNER_ADDRESS,
        addressList,
        {gasLimit: 6e6});
```

Reply:
> Bug: I think we supply 3 billion tokens, not 8. I suggest you put the supply in the constructor, not a factory parameter if possible.

This is old change, pls review new changes, this was updated here: https://github.com/moon-gaming/mee-governance-token/pull/14/files/a7626533a9a514c33bbba2c452a7808e4daaf7aa#diff-2cf9c5908dc8ce42e9ef5fcabb812f770e3648e84af9535aea9a331773e3e5b0R16